### PR TITLE
Re-enable block timeout calculation

### PR DESCRIPTION
**Pull request by igrr**
Block write timeout is generally determined by the time it takes to erase and write the data into flash. Since that time is proportional to the amount of uncompressed data written, we need to keep track of the number of bytes each chunk is decompressed into.

Re-enable the commented out code block, based on a suggestion (from Mark Adler, no less!) in https://stackoverflow.com/a/73096372.

## Testing

I've tested this by flashing a 2 MB sized empty SPIFFS image.

Without the fix, flashing times out.

With the fix, the timeout seems to be correctly calculated based on the uncompressed chunk size. With some debug logging added:
```
Compressed 2097152 bytes to 3825...
Debug: check_command enter compressed flash mode
Debug: Write loop 4096 0 1
Writing at 0x1000... (100%)
Debug: block compressed: 3825 uncompressed: 2097152
Debug: block timeout 83886.08
Debug: check_command write compressed data to flash after seq 0
Wrote 2097152 bytes (3825 compressed) at 0x1000 in 10.814 seconds.
```

* Replaces https://github.com/espressif/esptool-js/pull/57
* Fixes https://github.com/espressif/esptool-js/issues/49
* Fixes the remaining issue in https://github.com/espressif/esptool-js/issues/47

### CODE REVIEW

File: src/esploader.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { ESPError } from "./error";
-import { deflate } from "pako";
+import Pako, { deflate, Inflate } from "pako";
```
Comment (brianignacio5) [inline src/esploader.ts:2]: Suggest change imports to `import { Data, deflate, inflate } from "pako";`

File: src/esploader.ts [hunk 2]
```diff
@@ -935,32 +935,39 @@ export class ESPLoader {
       const t1 = d.getTime();
 
       let timeout = 5000;
+      // Create a decompressor to keep track of the size of uncompressed data
+      // to be written in each chunk.
+      const inflate = new Inflate({ chunkSize: 1 });
+      let total_len_uncompressed = 0;
+      inflate.onData = function (chunk: Pako.Data): void {
```
Comment (brianignacio5) [inline src/esploader.ts:942]: Also this to `chunk: Data`

**Review by brianignacio5 — CHANGES_REQUESTED** [review]
LGTM with few changes.

**Comment by balloob** [comment]
Could we have a new release once this is merged? 🙏

**Review by brianignacio5 — APPROVED** [review]
LGTM

**Comment by balloob** [comment]
🎉 

Would you have time for a release this week?

**Comment by balloob** [comment]
@igrr last time I'll ask, but could we get a release with this fix? Now that the repo has been migrated to TypeScript, we can no longer point our dependency at a git commit

**Comment by igrr** [comment]
@balloob I sorry for the delay! We have talked with @brianignacio5 about preparing the new release, but it seems like there is some issue in the release workflow that needs to be fixed, and neither of us has had time to look into this yet, unfortunately!

**Comment by Jason2866** [comment]
@balloob release is done :-)

**Pull request merged**